prov/tcp/src/xnet_srx.c

```diff
@@ -48,17 +48,28 @@ xnet_match_tag(struct xnet_srx *srx, struct xnet_ep *ep, uint64_t tag);
  */
 
 static struct xnet_xfer_entry *
-xnet_alloc_srx_xfer(struct xnet_srx *srx)
+xnet_alloc_srx_xfer(struct xnet_srx *srx, bool multi_recv)
 {
 	struct xnet_xfer_entry *xfer;
+	struct xnet_mrecv *mrecv = NULL;
 	struct xnet_progress *progress;
 
+	if (multi_recv) {
+		mrecv = calloc(1, sizeof(struct xnet_mrecv));
+		if (!mrecv)
+			return NULL;
+		mrecv->ref_cnt = 1;
+	}
```

Comment:
Could we do this allocation the first time we call xnet_alter_mrecv()?  That would leave the recv path unchanged and not impact performance when mrecv is not being used.  Most posted buffers will not have this flag set.